presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrderCurveSearcher.java

```diff
@@ -129,6 +112,45 @@ public static ZOrderCurveSearcher createZOrderCurveSearcher(ZOrder zOrder, List<
         return nonNullValues;
     }
 
+    private void combineAddressRanges(List<ZAddressRange<Long>> addressRanges)
+    {
+        if (addressRanges.isEmpty()) {
+            return;
+        }
+
+        addressRanges.sort((range1, range2) -> {
+            if (range1.equals(range2)) {
+                return 0;
+            }
+            if (range1.getMinimumAddress().equals(range2.getMinimumAddress())) {
+                return (range1.getMaximumAddress() > range2.getMaximumAddress()) ? 1 : -1;
+            }
+            return (range1.getMinimumAddress() > range2.getMinimumAddress()) ? 1 : -1;
+        });
+        List<ZAddressRange<Long>> tempRanges = addressRanges.stream().collect(ImmutableList.toImmutableList());
```

Comment:
Instead of creating an immutable copy and modifying the original, we should treat the original as immutable, and create a new list for the combined result